### DISCUSSION TITLE
Automatic releases?

### DISCUSSION THREAD

**Discussion by joshtriplett** [trillium-rs/trillium#471] [General]
Some projects, such as https://github.com/clap-rs/clap/ , have automated releases to the point that they're simple enough to release each crate whenever there's a change to it. This avoids the need to decide whether and when to release crates after merging PRs into them, and ensures that merged features or bugfixes get to users sooner.

Would you consider doing the same for trillium, to avoid having periods in which changes are available in current git but not available to users?

**Comment by jbr** [comment]
Not quite automatic, but I'm trying to use release-plz for releases. Unfortunately, this may have swung too far to the other extreme, as there's no way to release any workspace crate without releasing all workspace crates, including ones that are not really worth even a patch release. I'm likely going to either script my own bot or work on improving release-plz to support per-workspace-crate-prs

**Reply by joshtriplett** [reply]
@jbr Thank you for working on this! Honestly, having frequent patch releases seems like a net improvement. There's not much cost to a patch release, and if the patch numbers grow to the hundreds, that wouldn't be a big deal. :)

As long as release-plz avoids releasing a new version of a crate that has had *no* changes, it seems fine if it releases a crate that has had *minimal* changes.

**Reply by jbr** [reply]
It does seem like there's some unfortunate loop where it updates internal dependency versions, causing changes that it thinks need to be released. See https://github.com/trillium-rs/trillium/commit/1f2b6e28cde52dd54a851b51d4557e39d36cdd97

**Reply by joshtriplett** [reply]
@jbr It looks like it isn't deciding that it needs a new version solely from updating versions in `Cargo.toml`. It looks like it decides a new version if it updates `CHANGELOG.md`. :)

**Reply by joshtriplett** [reply]
By way of example, https://github.com/trillium-rs/trillium/pull/509 is really minor, but at the same time, if releases are automated, it's the kind of thing that'd be nice to have a patch release for. ;)

**Reply by jbr** [reply]
I was trying to release that with the merge of #503 (and did), but it brought along with it a bunch of noop patch releases. Opened https://github.com/MarcoIeni/release-plz/issues/1181 to reflect this
